jax/_src/numpy/lax_numpy.py

```diff
@@ -3647,8 +3655,17 @@ def _merge_static_and_dynamic_indices(treedef, static_idx, dynamic_idx):
       idx.append(s)
   return treedef.unflatten(idx)
 
-def _int(aval):
-  return not aval.shape and issubdtype(aval.dtype, integer)
+def _is_basic_int_index(x):
+  if isinstance(x, core.Tracer):
+    aval = x.aval
+    return (isinstance(aval, (ConcreteArray, ShapedArray)) and
```

Comment:
`ConcreteArray` is a subclass of `ShapedArray`, so `isinstance(aval, ShapedArray)` should be sufficient.

Reply:
Updated in bb518728a576a67f3e4ff9c48552dd58600b8df7